pkg/prom/ha/sharding.go

```diff
@@ -22,6 +23,11 @@ import (
 //
 // Satisfies agentproto.ScrapingServiceServer.
 func (s *Server) Reshard(ctx context.Context, _ *agentproto.ReshardRequest) (_ *empty.Empty, err error) {
+	l, ctx := spanlogger.NewFromLogger(ctx, s.logger, "Server.Reshard")
+	defer l.Finish()
+
+	level.Info(l).Log("msg", "resharding agent")
```

Comment:
The "resharding agent" message was moved here to be able to know when Reshard is called in all circumstances. Before, it was only logged for local resharding on the resharding interval, but wasn't logged when a remote Agent invokes the Reshard gRPC method.